### PR TITLE
Update to ASM 9.6 + POM and code refresh

### PR DESCRIPTION
**Pull request by rfelcman**
There is output from `org.eclipse.dash:license-tool-plugin`

```
maven/mavencentral/org.ow2.asm/asm-commons/9.6, BSD-3-Clause, approved, #10775
maven/mavencentral/org.ow2.asm/asm-tree/9.6, BSD-3-Clause, approved, #10773
maven/mavencentral/org.ow2.asm/asm/9.6, BSD-3-Clause, approved, #10776
```

### CODE REVIEW

File: org.eclipse.persistence.asm/pom.xml [hunk 1]
```diff
@@ -199,17 +199,17 @@
...
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-javadoc-plugin</artifactId>
-                    <version>3.5.0</version>
+                    <version>3.6.0</version>
                 </plugin>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-source-plugin</artifactId>
-                    <version>3.2.1</version>
+                    <version>3.3.0</version>
```
Comment (lukasj) [inline org.eclipse.persistence.asm/pom.xml:207]: I'm afraid this version will fail the build on eclipse infra (at least I'm experiencing that in few projects)

**Comment by rfelcman** [comment]
JDK 21 test fails due eclipse.dash tool. Local build works.

**Comment by lukasj** [comment]
dash invocation can be moved to an extra step (and it will run just once..)

**Comment by rfelcman** [comment]
> dash invocation can be moved to an extra step (and it will run just once..)

It works now.

**Pull request merged**